src/System.Net.Security/src/System/Net/Unix/SecuritySafeHandles.cs

```diff
@@ -19,24 +19,32 @@ internal sealed class SafeFreeCertContext : DebugSafeHandle
     internal sealed class SafeFreeCertContext : SafeHandle
     {
 #endif
-        public SafeFreeCertContext(IntPtr handle) : base(handle, true)
+        private readonly SafeX509Handle _certificate;
+
+        public SafeFreeCertContext(SafeX509Handle certificate) : base(IntPtr.Zero, true)
         {
+            if ((null != certificate) && !certificate.IsInvalid)
```

Comment:
When server queries for the client certificate, it may be null and that is ok. Added a comment